File: python_backend/getters/get_piece_types.py
from custom_except import *
from getters.get_piece_type import get_piece_type


def get_piece_types(board):
    """get the type of piece of every piece instance on board. sort alphabetically and return"""
    piece_ids = list(filter(lambda id_: id_ != '#', board.values()))
    piece_types = list(map(lambda id_: get_piece_type(id_), piece_ids))
    piece_types.sort()
    return piece_types

File: node.js/pieceType/getPieceTypes.js
var  getPieceType  = require("../pieceType/getPieceType");


function getPieceTypes(board) {
    /*get the type of piece of every piece instance on board. sort alphabetically &&  return**/
    let pieceIds = Object.values(board).filter(id => id !== '#')
    let pieceTypes = pieceIds.map(id => getPieceType(id))
    pieceTypes.sort();
    return pieceTypes
}
    
module.exports = getPieceTypes;